### ISSUE TITLE
Inheriting class with __slots__ defined

### ISSUE THREAD

**Issue by DenisOH** [python/mypy#8358]
* Are you reporting a bug, or opening a feature request?

Feature request

* Please insert below the code you are checking with mypy.

```
class Base():
    __slots__: List[str] = ['foo', 'bar']
    if TYPE_CHECKING:
        foo: int
        bar: int

class Derived(Base):
    __slots__: List[str] = ['bar']
```

* What is the actual behavior/output?

`Success: no issues found in 1 source file`

* What is the behavior/output you expect?

I believe it would be nice if mypy generated an error on `__slots__` declaration in Derived class because 'bar' slot was already declared in Base class. 

* What are the versions of mypy and Python you are using?

Python 3.8.0, mypy 0.761

**Comment by JukkaL** [comment]
The redundant slot definition seems to work at runtime, so this seems pretty low priority.

**Comment by DenisOH** [comment]
@JukkaL Agree. On a side note, having the redundant slot wastes some memory.
